File: app/src/main/java/ms/imagine/foodiemate/presenters/FbDatabaseWrite.java
package ms.imagine.foodiemate.presenters;

import android.util.Log;
import ms.imagine.foodiemate.callbacks.DbWriteCallBacks;
import ms.imagine.foodiemate.data.Eggs;

import java.util.HashMap;
import java.util.Map;

class FbDatabaseWrite extends FbDatabasePresenter{
    private DbWriteCallBacks callback;

    FbDatabaseWrite(String _uid, DbWriteCallBacks callBacks) {
        super(_uid);
        callback = callBacks;
    }

    public void writeEgg(Eggs eggs){
        Map<String, Object> map = new HashMap<>();
        map.put("eggTag", eggs.getEggtag());
        map.put("timestamp", eggs.getTimestamp());
        map.put("status", eggs.getStatus());
        map.put("imgURL", eggs.getRemoteImgURL());

        if (!eggs.isLegacyEgg()){
            map.put("egglist", eggs.getEgg());
        }

        String leKey = firebaseDB.push().getKey();
        firebaseDB.child(leKey).updateChildren(map)
        .addOnCompleteListener((s)->callback.hashCode())
                .addOnFailureListener(e -> {
                    Log.d("", "", e);});
    }
}

File: app/src/main/java/ms/imagine/foodiemate/Presenter/FbDatabaseWrite.kt
package ms.imagine.foodiemate.Presenter

import ms.imagine.foodiemate.callbacks.DbWriteCallBacks
import ms.imagine.foodiemate.data.Eggs

class FbDatabaseWrite: FbDatabasePresenter{
    var callback: DbWriteCallBacks

    constructor(_uid:String?, callBacks: DbWriteCallBacks) :
            super(_uid) {
        callback = callBacks
    }

    fun writeEgg(eggs: Eggs){
        var map = HashMap<String, Any>()
        map.put("eggTag", eggs.eggtag)
        map.put("timestamp", eggs.timestamp)
        map.put("status", eggs.status)
        map.put("imgURL", eggs.remoteImgURL)

        if (!eggs.isLegacyEgg()){
            map.put("egglist", eggs.egg)
        }

        var leKey = firebaseDB.push().key
        firebaseDB.child(leKey).updateChildren(map as Map<String, Any>?)
                .addOnCompleteListener { callback.hashCode() }
                .addOnFailureListener {  }
    }
}